src/cli.rs

```diff
@@ -1,55 +1,79 @@
-use crate::dependency::{Dependency, parse_dependency};
+use crate::dependency::Dependency;
 
-/// This tool allow you to create a new Rust temporary project in a temporary
-/// directory.
+/// cargo-temp: Create temporary Rust projects with specified dependencies.
 ///
-/// The dependencies can be provided in arguments (e.g.`cargo-temp anyhow
-/// tokio`). When the shell is exited, the temporary directory is deleted unless
-/// you removed the file `TO_DELETE`.
+/// Cargo subcommand for quick experimentation. It creates a temporary project, adds any
+/// dependencies you specify and drops you into a shell.
 #[derive(clap::Parser, Debug, Clone)]
 #[command(author, version, about, long_about)]
 pub struct Cli {
-    /// Dependencies to add to `Cargo.toml`.
-    ///
-    /// The default version used is `*` but this can be replaced using `=`.
-    /// E.g. `cargo-temp anyhow=1.0.13`
-    #[arg(value_parser = parse_dependency)]
+    /// Dependencies to add to the project's `Cargo.toml`.
+    #[arg(long_help(
+        "Each DEPENDENCY can take one of the following forms:
+
+    (<NAME> | <URL>[#<BRANCH> | <REV>])[=<VERSION>][+][+<FEATURE>...]
+
+You must provide either a `NAME` (e.g. `anyhow`) or a `URL` pointing to a git repository. URLs can
+use `http(s)` or `ssh` schemes and may include a branch or a revision using `#`.
+
+You can optionally specify a version with `=` (e.g. `tokio=1.48`). Operators following cargo's
+comparison requirements can also be provided after the `=`:
+
+- `=`: Exact version (e.g. `tokio==1.48`).
+- `>`: Maximal version (e.g. `tokio>1.48`).
+- `<`: Minimal version (e.g. `tokio<1.48`).
+- `>=`: Maximal or equal version (e.g. `tokio>=1.48`).
+- `<=`: Minimal or equal version (e.g. `tokio<=1.48`).
+- `~`: Minimal version with some ability to update (e.g. `tokio~1`).
+
+Features can be enabled by appending them with `+` (e.g. `clap+derive` or `clap+derive+cargo`. To
+disable default features, prefix the first feature with an additional `+`(e.g. `ratatui++termion`
+or `ratatui++termion+serde`).
+
+# Examples
+
+```sh
+cargo temp anyhow
+cargo temp tokio=1.48
+cargo temp clap+derive
+cargo temp https://github.com/rust-random/rand#thread_rng
+cargo temp ssh://git@github.com/ratatui/ratatui.git#latest++termion
+```"
+    ))]
     pub dependencies: Vec<Dependency>,
 
-    /// Create a library instead of a binary.
+    /// Create a library project instead of a binary.
     #[arg(long, short = 'l')]
     pub lib: bool,
 
+    /// Select the Rust edition for the temporary project.
+    ///
+    /// Default to the latest stable edition. Possible values are the same as `cargo new --edition`.
+    #[arg(long, short = 'e')]
+    pub edition: Option<String>,
```

Comment:
Come to think of it, the edition is probably not very helpful. Maybe the rust version would be very helpful but this might already be handled with +1.65 in first argument before "temp"